### PR TITLE
merge kubeflow/common.v1 to mpi-operator

### LINKED ISSUE #564: Copy APIs from common repo into here
We merged the common repo into the training operator. So we will archive the common repo.
Hence, coping APIs from common into here would be good.

ref: https://github.com/kubeflow/training-operator/pull/1813

### PR DESCRIPTION
**Pull request by lowang-bh**
Fix #564 

TODO:

  - [x] `hack/python-sdk/gen-sdk.sh` did not finished due to lack of java runtime.

```sh
hack/python-sdk/gen-sdk.sh
Generating V2 OpenAPI specification ...
Generating V2 swagger file ...
Downloading the swagger-codegen JAR package ...
Generating V2 Python SDK for Kubeflow MPI-Operator ...
The operation couldn’t be completed. Unable to locate a Java Runtime.
Please visit http://www.java.com for information on installing Java.
```

### CODE REVIEW

**Comment by lowang-bh** [comment]
@terrytangyuan @tenzen-y would you please take a look at this pr? thanks.

File: pkg/apis/kubeflow/v2beta1/constants.go [hunk 1]
```diff
@@ -14,15 +14,32 @@
...
-	DefaultRestartPolicy = common.RestartPolicyNever
+	DefaultRestartPolicy = RestartPolicyNever
 	// DefaultLauncherRestartPolicy is default RestartPolicy for Launcher Job.
-	DefaultLauncherRestartPolicy = common.RestartPolicyOnFailure
+	DefaultLauncherRestartPolicy = RestartPolicyOnFailure
 	// OperatorName is the name of the operator used as value to the label common.OperatorLabelName
 	OperatorName = "mpi-operator"
 )
+
+// merge from common.v1
```
Comment (tenzen-y) [inline pkg/apis/kubeflow/v2beta1/constants.go:28]: Can you leave a link to the common repo here?

Comment (lowang-bh) [reply pkg/apis/kubeflow/v2beta1/constants.go:28]: done.

**Review by tenzen-y — COMMENTED** [review]
@lowang-bh Overall LGTM

File: pkg/apis/kubeflow/v2beta1/types.go [hunk 2]
```diff
@@ -314,3 +313,45 @@ const (
...
+	RestartPolicyOnFailure RestartPolicy = "OnFailure"
+	RestartPolicyNever     RestartPolicy = "Never"
+
+	// RestartPolicyExitCode policy means that user should add exit code by themselves,
+	// The job operator will check these exit codes to
+	// determine the behavior when an error occurs:
+	// - 1-127: permanent error, do not restart.
+	// - 128-255: retryable error, will restart the pod.
+	RestartPolicyExitCode RestartPolicy = "ExitCode"
+)
```
[the remark this answers is not in the document]
Comment (tenzen-y) [reply pkg/apis/kubeflow/v2beta1/types.go:357]:
```shell
...
################################################################################
# Thanks for using OpenAPI Generator.                                          #
# Please consider donation to help us maintain this project 🙏                 #
# https://opencollective.com/openapi_generator/donate                          #
################################################################################
Kubeflow MPI-Operator Python SDK is generated successfully to folder sdk/python/.
git --no-pager diff --exit-code manifests/base deploy sdk pkg/apis pkg/client
diff --git a/pkg/apis/kubeflow/v2beta1/types.go b/pkg/apis/kubeflow/v2beta1/types.go
index 8247938..bffd1b5 100644
--- a/pkg/apis/kubeflow/v2beta1/types.go
+++ b/pkg/apis/kubeflow/v2beta1/types.go
@@ -354,4 +354,4 @@ const (
 	// - 1-127: permanent error, do not restart.
 	// - 128-255: retryable error, will restart the pod.
 	RestartPolicyExitCode RestartPolicy = "ExitCode"
-)
\ No newline at end of file
+)
make: *** [Makefile:94: verify-generate] Error 1
Error: Process completed with exit code 2.
```

**Comment by tenzen-y** [comment]
@lowang-bh Can you fix CI errors?


```shell
?   	github.com/kubeflow/mpi-operator/pkg/version	[no test files]
# github.com/kubeflow/mpi-operator/test/integration [github.com/kubeflow/mpi-operator/test/integration.test]
Error: test/integration/mpi_job_controller_test.go:734:50: undefined: common
FAIL	github.com/kubeflow/mpi-operator/test/integration [build failed]
FAIL
make: *** [Makefile:69: test] Error 2
Error: Process completed with exit code 2.
```

https://github.com/kubeflow/mpi-operator/actions/runs/5462212552/jobs/9941258951?pr=571#step:8:277

```shell
...
go: downloading github.com/prometheus/procfs v0.7.3
go: downloading github.com/matttproud/golang_protobuf_extensions v1.0.2-0.20181231171920-c182affec369
# github.com/kubeflow/mpi-operator/test/e2e [github.com/kubeflow/mpi-operator/test/e2e.test]
Error: test/e2e/mpi_job_test.go:460:7: undefined: common
FAIL	github.com/kubeflow/mpi-operator/test/e2e [build failed]
```

https://github.com/kubeflow/mpi-operator/actions/runs/5462212552/jobs/9941259352?pr=571#step:4:3326

**Review by tenzen-y — COMMENTED** [review]
LGTM
/assign @alculquicondor

**Comment by tenzen-y** [comment]
cc: @alculquicondor

**Review by alculquicondor — COMMENTED** [review]
/approve

**Review by terrytangyuan — APPROVED** [review]
Thanks!

/lgtm

**Comment by tenzen-y** [comment]
@terrytangyuan @alculquicondor Could you restart `build / E2E (v1.27.1) (pull_request)` ?

**Comment by terrytangyuan** [comment]
Done

**Pull request merged**